### PR TITLE
Create Block: Speed up scaffolding by omitting WordPress dependencies

### PR DESCRIPTION
**Pull request by gziolo**
## Description

Follow-up for #27880.

**TLDR; Omit verification for WordPress dependencies in the import statements since they get externalized when used with WordPress.**

In #27880, I added an option to provide a list of npm dependencies to install when scaffolding the block with `@wordpress/create-block`. The rationale behind it was summarized as follows:

> - in development, those dependencies are necessary for linting and unit testing
> - Visual Studio Code and other IDEs use those packages for code completion and all the hints
> - in general, as a good practice `package.json` should list all production npm dependencies used in the project

In practice, it slows down the scaffolding process too much. It also created some follow-up discussion on whether we could skip linting for WordPress packages that are handled as externals anyway.

This PR explores changes to the ESLint Plugin which ignores all WordPress packages in the linting process for import statements. The exception is disabled in the Gutenberg plugin so we can ensure that npm packages are always correctly configured for usage from npm.

As part of the effort, I think we should remove WordPress packages from the listed npm dependencies in templates so they install way faster. In addition to that, I added a check for when `wp-scripts` gets explicitly disabled to skip npm dependencies because they aren't that useful in such cases.

## How has this been tested?

I executed ` ./bin/test-create-block.sh ` to ensure that linting still passes for the scaffolded block with the default `esnext` template even when there are no WordPress dependencies installed.

I added a test dependency (`redux` that existed in `node_modules`) in the scaffolded template to ensure that the linting rule still throws errors for non-WordPress dependencies:

```diff
diff --git a/packages/create-block/lib/templates/esnext/src/edit.js.mustache b/packages/create-block/lib/templates/esnext/src/edit.js.mustache
index 8ed2bb5d38..24f4624aaf 100644
--- a/packages/create-block/lib/templates/esnext/src/edit.js.mustache
+++ b/packages/create-block/lib/templates/esnext/src/edit.js.mustache
@@ -4,6 +4,7 @@
  * @see https://developer.wordpress.org/block-editor/packages/packages-i18n/
  */
 import { __ } from '@wordpress/i18n';
+import 'redux';
 
 /**
  * React hook that is used to mark the block wrapper element.
```

Screenshot 2021-12-28 at 12 56 06

## Types of changes

Enhancement.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [-] My code follows the accessibility standards. 
- [-] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/eslint-plugin/configs/recommended-with-formatting.js [hunk 1]
```diff
@@ -20,14 +20,23 @@ const config = {
...
 			'error',
 			{
 				peerDependencies: true,
 			},
 		],
-		'import/no-unresolved': 'error',
+		'import/no-unresolved': [
+			'error',
+			{
+				ignore: [ '^@wordpress/' ],
```
Comment (fabiankaegy) [inline packages/eslint-plugin/configs/recommended-with-formatting.js]: Is there a way in which we can exclude packages like the '@wordpress/icons' package from this ignore lists? It already is confusing which ones are included in core and which ones aren't.

**Review by fabiankaegy — APPROVED** [review]
I left one question regarding some of the packages that are not available through the window object in core but besides that this is looking good to me and I think the speed improvement is really worth it.

File: packages/eslint-plugin/configs/recommended-with-formatting.js (hunk 1 above)
Comment (gziolo) [reply packages/eslint-plugin/configs/recommended-with-formatting.js]: Good catch. It's also `@wordpress/interface`. I completely forgot about them. I'm positive we can tweak the regular expression 👍🏻

Comment (gziolo) [reply packages/eslint-plugin/configs/recommended-with-formatting.js]: https://github.com/WordPress/gutenberg/pull/37639/commits/2360498eebb4e1afc1b47e4634d8b0e4c856241f should do the trick.

File: packages/eslint-plugin/configs/recommended-with-formatting.js [hunk 2]
```diff
@@ -3,6 +3,8 @@
  */
 const { isPackageInstalled } = require( '../utils' );
 
+const wpPackagesRegExp = '^@wordpress/(?!(icons|interface))';
```
Comment (gziolo) [inline packages/eslint-plugin/configs/recommended-with-formatting.js:7]:
```suggestion
// Exclude bundled WordPress packages from the list.
const wpPackagesRegExp = '^@wordpress/(?!(icons|interface))';
```
